odoo/sql_db.py

```diff
@@ -114,24 +170,9 @@ def __init__(self):
     @check
     def savepoint(self, flush=True):
         """context manager entering in a new savepoint"""
-        name = uuid.uuid1().hex
-        if flush:
-            flush_env(self, clear=False)
-            self.precommit.run()
-        self.execute('SAVEPOINT "%s"' % name)
-        try:
-            yield
-            if flush:
-                flush_env(self, clear=False)
-                self.precommit.run()
-        except Exception:
-            if flush:
-                clear_env(self)
-                self.precommit.clear()
-            self.execute('ROLLBACK TO SAVEPOINT "%s"' % name)
-            raise
-        else:
-            self.execute('RELEASE SAVEPOINT "%s"' % name)
+        flusher = flushing(self) if flush else nullcontext()
+        with Savepoint(self) as sp, flusher: # pylint: disable=confusing-with-statement
```

Comment:
I understand why pylint complains that it's confusing, you could simply add a comment above explaining that here we enter both the `Savepoint` and the `flushing` contexts, the latter only if `flush=True`, or you could add it / update the docstring

Reply:
I've no idea what you're suggesting. A comment would not suppress pylint's dumbassery.

The docstring is irrelevant, this is internal behaviour (identical to the old one... except the flush is now performed after entering the savepoint which might be an issue, bah, humbug, my nice conditional contextmanager doesn't work, I'll have to move the setup outside of the contextmanager to ensure the nesting is correct).

Reply:
I mean I understand why pylint thinks it's a confusing statement, I'm just suggesting you add a comment to explain what's going on in case some future reader does find it confusing, all while keeping the disable directive ofc

Reply:
Updated `flushing` to be composing so it performs its flushes at the correct point (relative to the savepoint), which leads to not needing the pylint suppression (or the nullcontext) anymore.